src/entity_embedding_utils.jl

```diff
@@ -0,0 +1,148 @@
+    for featname in specified_featnames
+        if embedding_dims[featname] isa AbstractFloat
+            embedding_dims[featname] = ceil(
+                Int,
+                embedding_dims[featname] *
+                catind2numlevels[findfirst(x -> x == featname, featnames)],
+            )
+        end
+    end
+    newdims = [
+        (cat_ind in specified_featinds) ? embedding_dims[featnames[cat_ind]] :
+        set_default_new_embedding_dim(num_levels[i]) for
+        (i, cat_ind) in enumerate(cat_inds)
+    ]
+    return newdims
+end
+
+
+"""
+**Private Method**
+
+Returns the indices of the categorical columns in the table `X`.
+"""
+function get_cat_inds(X)
+    # if input is a matrix; conclude no categorical columns
+    Tables.istable(X) || return Int[]
+    types = schema(X).scitypes
+    cat_inds = findall(x -> x <: Finite, types)
+    return cat_inds
+end
+
+"""
+**Private Method**
+
+Returns the number of levels in each categorical column in the table `X`.
+"""
+function get_num_levels(X, cat_inds)
+    num_levels = []
+    for i in cat_inds
+        num_levels = push!(num_levels, length(levels(Tables.getcolumn(X, i))))
+    end
+    return num_levels
+end
+
+# A function to prepare the inputs for entity embeddings layer
+function prepare_entityembs(X, featnames, cat_inds, embedding_dims)
+    # 1. Construct entityprops
+    numfeats = length(featnames)
+    num_levels = get_num_levels(X, cat_inds)
+    newdims = set_new_embedding_dims(featnames, cat_inds, num_levels, embedding_dims)
+    entityprops = [
+        (index = cat_inds[i], levels = num_levels[i], newdim = newdims[i]) for
+        i in eachindex(cat_inds)
+    ]
+    # 2. Compute entityemb_output_dim
+    entityemb_output_dim = sum(newdims) + numfeats - length(cat_inds)
+    return entityprops, entityemb_output_dim
+end
+
+# A function to construct model chain including entity embeddings as the first layer
+function construct_model_chain_with_entityembs(
+    model,
+    rng,
+    shape,
+    move,
+    entityprops,
+    entityemb_output_dim,
+)
+    chain = try
+        Flux.Chain(
+            EntityEmbedder(entityprops, shape[1]; init = Flux.glorot_uniform(rng)),
+            build(model, rng, (entityemb_output_dim, shape[2])),
+        ) |> move
+    catch ex
+        @error ERR_BUILDER
+        rethrow()
+    end
+    return chain
+end
+
+
+# A function that given a model chain, returns a dictionary of embedding matrices
+function get_embedding_matrices(chain, cat_inds, featnames)
+    embedder_layer = chain.layers[1]
+    embedding_matrices = Dict{Symbol, Matrix{Float32}}()
+    for cat_ind in cat_inds
+        featname = featnames[cat_ind]
+        matrix = Flux.params(embedder_layer.embedders[cat_ind])[1]
+        embedding_matrices[featname] = matrix
+    end
+    return embedding_matrices
+end
+
+
+
+# Transformer for entity-enabled models
+function MLJModelInterface.transform(
+    transformer::EMBEDDING_ENABLED_MODELS_UNION,
```

Comment:
```suggestion
    transformer
```

Instead, just throw an error if `is_embedding_enabled(model)` is `true`, or return `X` .

Reply:
Done in e5d8141fd2faac36619fdfb72d452960283d4d95